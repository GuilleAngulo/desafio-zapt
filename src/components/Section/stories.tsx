import { Story, Meta } from '@storybook/react'
import Section, { SectionProps } from '.'

export default {
  title: 'Section',
  component: Section,
  args: {
    title: 'This is a section',
    children: <p>Content</p>
  }
} as Meta

export const Default: Story<SectionProps> = (args) => <Section {...args} />
