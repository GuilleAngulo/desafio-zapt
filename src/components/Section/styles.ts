import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    padding: ${theme.spacings.small} ${theme.spacings.xsmall};

    ${media.greaterThan('small')`
      padding: ${theme.spacings.large} ${theme.spacings.xlarge};
    `}
    ${media.greaterThan('large')`
      padding: ${theme.spacings.xxlarge} ${theme.spacings.huge};
    `}
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    font-weight: ${theme.font.light};
    font-size: ${theme.font.sizes.xlarge};
    text-align: center;
    margin-bottom: ${theme.spacings.xlarge};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.huge};
    `}
  `}
`
