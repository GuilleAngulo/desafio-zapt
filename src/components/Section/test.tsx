import { render, screen } from 'utils/test-utils'

import Section from '.'

describe('<Section />', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Section title="Section Title">
        <p>Content</p>
      </Section>
    )

    expect(
      screen.getByRole('heading', { name: /section title/i })
    ).toBeInTheDocument()

    expect(screen.getByText(/content/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
