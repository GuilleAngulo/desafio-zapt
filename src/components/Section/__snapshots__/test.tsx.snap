// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should render correctly 1`] = `
.c0 {
  padding: 1.6rem 1.2rem;
}

.c1 {
  font-weight: 300;
  font-size: 2.8rem;
  text-align: center;
  margin-bottom: 4.8rem;
}

@media (min-width:450px) {
  .c0 {
    padding: 3.2rem 4.8rem;
  }
}

@media (min-width:1170px) {
  .c0 {
    padding: 8rem 12rem;
  }
}

@media (min-width:768px) {
  .c1 {
    font-size: 4rem;
  }
}

<section
  class="c0"
>
  <h2
    class="c1"
  >
    Section Title
  </h2>
  <p>
    Content
  </p>
</section>
`;
